File: app/src/main/java/com/geridea/trentastico/model/cache/StudyCourseNotCachedInterval.java
package com.geridea.trentastico.model.cache;


/*
 * Created with ♥ by Slava on 27/03/2017.
 */

import com.geridea.trentastico.network.controllers.LessonsController;
import com.geridea.trentastico.network.controllers.listener.LessonsLoadingListener;
import com.geridea.trentastico.utils.time.WeekInterval;
import com.geridea.trentastico.utils.time.WeekTime;

public class StudyCourseNotCachedInterval extends NotCachedInterval {

    public StudyCourseNotCachedInterval(WeekTime start, WeekTime end) {
        super(start, end);

    }

    @Override
    public void launchLoading(LessonsController controller, LessonsLoadingListener listener) {
        controller.sendStudyCourseLoadingRequest(this, listener);
    }

    @Override
    public void launchLoadingOneTime(LessonsController controller, LessonsLoadingListener listener) {
        controller.sendStudyCourseLoadingRequestOneTime(this, listener);
    }

    public StudyCourseNotCachedInterval(WeekInterval interval) {
        super(interval.getStartCopy(), interval.getEndCopy());
    }

}

File: app/src/main/java/com/geridea/trentastico/model/cache/StudyCourseNotCachedInterval.kt
package com.geridea.trentastico.model.cache


/*
 * Created with ♥ by Slava on 27/03/2017.
 */

import com.geridea.trentastico.network.controllers.LessonsController
import com.geridea.trentastico.network.controllers.listener.LessonsLoadingListener
import com.geridea.trentastico.utils.time.WeekInterval

class StudyCourseNotCachedInterval(interval: WeekInterval) : NotCachedInterval(interval.startCopy, interval.endCopy) {

    override fun launchLoading(controller: LessonsController, listener: LessonsLoadingListener) {
        controller.sendStudyCourseLoadingRequest(this, listener)
    }

    override fun launchLoadingOneTime(controller: LessonsController, listener: LessonsLoadingListener) {
        controller.sendStudyCourseLoadingRequestOneTime(this, listener)
    }

}
